teams_rules_test.go

```diff
@@ -224,6 +246,23 @@ func TestTeamsCreateRule(t *testing.T) {
 
 	handler := func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal(t, http.MethodPost, r.Method, "Expected method 'POST', got %s", r.Method)
+		type PartialTeamsCheckSessionSettings struct {
```

Comment:
why are we recreating structs here? should you be using the ones you created above?

Reply:
I didn't want to reuse the ones above because the main thing I wanted to test was the `duration` is being correctly serialized into that specific string. If I reused the structs above, I would only be testing that we can deserialize what we serialize, not that it gets serialized to this format.

Reply:
Maybe I'm just being too conservative, as we already testing the deserialization from `15m0s` into `Duration` is correct, and there's no reason the serialization wouldn't be also. If you think so too, I don't mind removing this block of serialization validations I added on the current PR.

Reply:
that seems fine to me 👍